botorch/models/higher_order_gp.py

```diff
@@ -502,8 +504,10 @@ def posterior(
             )
             if hasattr(self, "outcome_transform"):
                 posterior = self.outcome_transform.untransform_posterior(posterior)
-
-            return posterior
+            if posterior_transform is not None:
+                return posterior_transform(posterior)
```

Comment:
Hmm do we need to be raising an exception in `botorch.posteriors.gpytorch.scalarize_posterior` for HOGP?

Reply:
Hmm yeah let me do this for now to be safe. May want to think through this some more down the road. cc @wjmaddox

Reply:
What's the current error with the HOGP and these types of transforms?

Reply:
I think I'll just raise a `NotImplementedError` here if `posterior_transform` is not `None` for now.

Reply:
> What's the current error with the HOGP and these types of transforms?

Not sure if it technically raises an error, but basically `scalarize_posterior` assumes a standard `GPyTorchPosterior` and will always return a `GPyTorchPosterior` and in some cases may try to instantiate the covariance. I don't think we have a use case for this at this point so maybe it's just safer to error out and revisit this when needed.

Reply:
Yeah, that probably should be applied to `KroneckerMultitaskGP` then as well for the same reason (forming covariances is expensive).

Reply:
Addressed in #1076